ports/views.py

```diff
@@ -0,0 +1,293 @@
+from django.shortcuts import render
+from django.http import HttpResponse
+from django.views.decorators.csrf import csrf_exempt
+from django.core.exceptions import ObjectDoesNotExist
+from .models import Port, Category, BuildHistory, Maintainer, Dependency, Builder, User, Variant
+from bs4 import BeautifulSoup
+from django.core.paginator import Paginator, PageNotAnInteger, EmptyPage
+import requests, json, datetime
```

Comment:
just to keep all the comments at one place, I'm pasting the link here again: https://www.python.org/dev/peps/pep-0008/#imports